### DISCUSSION TITLE
How insert block from another dwg file?

### DISCUSSION THREAD

**Discussion by davi-blsoftwares** [DomCR/ACadSharp#372] [Q&A] [answered]
I would like insert blocks from another dwg files.
In AutoCAD i use this method:

/// <summary>
        /// Carrega bloco no dwg
        /// </summary>
        /// <param name="blockFileName">Caminho completo do bloco</param>
        /// <param name="blockName">Nome do bloco</param>
        /// <param name="blockId">Retorna o ObjectId da definição do bloco</param>
        /// <returns>Retorna true se o bloco foi carregado senão false</returns>
        public static bool loadBlock(string blockFileName, string blockName, ref ObjectId blockId)
        {
            blockName = blockName.Trim();
            try
            {
                Database curDb = Application.DocumentManager.MdiActiveDocument.Database;
                Database blockDb = new Database(false, true);

#if ZWCAD
                var modo = FileShare.Read;
#else
                var modo = FileOpenMode.OpenForReadAndAllShare;
#endif

                blockDb.ReadDwgFile(blockFileName, modo, true, "");
                blockId = curDb.Insert(blockName, blockDb, true);

                return true;
            }
            catch (System.Exception e)
            {
                erros.AppendLine(e.Message);
                return false;
            }
            // }
        }
How make this in AcadSharp?

**Comment by DomCR** [comment]
Hi @davi-blsoftwares 

To import a block record from another document you can use this code as an example:

```C#
	//Read the document
	CadDocument doc;
	using (DwgReader reader = new DwgReader(_file))
	{
		doc = reader.Read();
	}

	//Create a new document or read another one
	CadDocument transfer = new CadDocument();
	//Create a clone of the block that you want to import to the new document
	BlockRecord block = (BlockRecord)doc.BlockRecords["MyBlock"].Clone();
	transfer.BlockRecords.Add(block);

	//Save the changes
	using (DwgWriter writer = new DwgWriter(outPath, transfer))
	{
		writer.Write();
	}
```

While using this code I've noticed an error in the clone method, I've open a branch to fix it.

If you are using the Nuget Package please let me know and I'll make a release once is fixed.

Hope this helps!

**Reply by DomCR** [reply, marked as answer]
- https://github.com/DomCR/ACadSharp/pull/373
